consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final String DEFAULT_SEPARATOR =  "\t";;
+
+    private static final String END_OF_LINE = System.lineSeparator();
+
+    public void setApprovedDARHeader(FileWriter darWriter) throws IOException {
+        darWriter.write(
+                HeaderDAR.DAR_ID.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATASET_NAME.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATASET_ID.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.CONSENT_ID.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATA_REQUESTER_NAME.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.ORGANIZATION.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.CODED_VERSION_SDUL.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.CODED_VERSION_DAR.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.RESEARCH_PURPOSE.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATE_REQUEST_SUBMISSION.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATE_REQUEST_APPROVAL.getValue() + DEFAULT_SEPARATOR +
+                        HeaderDAR.DATE_REQUEST_RE_ATTESTATION.getValue() + END_OF_LINE);
+    }
+
+    public void addApprovedDARLine(FileWriter darWriter, Election election, Document dar, String profileName, String institution, Consent consent) throws IOException {
+        List<Document> dataSetDetail = dar.get(DarConstants.DATASET_DETAIL, ArrayList.class);
+        String dataSetName = CollectionUtils.isNotEmpty(dataSetDetail) ? dataSetDetail.get(0).getString("name") : " ";
+        String sDUL = StringUtils.isNotEmpty(consent.getTranslatedUseRestriction()) ?  consent.getTranslatedUseRestriction().replace("\n", " ") : "";
+        String translatedRestriction = StringUtils.isNotEmpty(dar.getString(DarConstants.TRANSLATED_RESTRICTION)) ? dar.getString(DarConstants.TRANSLATED_RESTRICTION).replace("<br>", " ") :  "";
+        String rusSummary = StringUtils.isNotEmpty( dar.getString(DarConstants.NON_TECH_RUS)) ?  dar.getString(DarConstants.NON_TECH_RUS).replace("\n", " ") : "";
+        darWriter.write(
+                dar.getString(DarConstants.DAR_CODE) + DEFAULT_SEPARATOR +
+                dataSetName + DEFAULT_SEPARATOR +
+                dar.get(DarConstants.DATASET_ID, ArrayList.class).get(0).toString() + DEFAULT_SEPARATOR +
+                consent.getName() + DEFAULT_SEPARATOR +
+                profileName + DEFAULT_SEPARATOR +
+                institution + DEFAULT_SEPARATOR +
+                sDUL + DEFAULT_SEPARATOR +
+                translatedRestriction + DEFAULT_SEPARATOR +
+                rusSummary + DEFAULT_SEPARATOR +
+                formatTimeToDate(dar.getDate(DarConstants.SORT_DATE).getTime()) + DEFAULT_SEPARATOR +
+                formatTimeToDate(election.getFinalVoteDate().getTime()) + DEFAULT_SEPARATOR +
+                " " + END_OF_LINE);
+    }
```

Comment:
i feel like you could combine the common functionality from this and addReviewedDARLine in a single method and call it from the specific methods.

something like:
```java
private void addDARLine(FileWriter darWriter, Election election, Document dar, String customContent1, String customContent2, Consent consent) throws IOException {
        List<Document> dataSetDetail = dar.get(DarConstants.DATASET_DETAIL, ArrayList.class);
        String sDUL = StringUtils.isNotEmpty(election.getTranslatedUseRestriction()) ? election.getTranslatedUseRestriction() : consent.getTranslatedUseRestriction();
        sDUL = StringUtils.isNotEmpty(sDUL) ?  sDUL.replace("\n", "") : "";
        String translatedRestriction = StringUtils.isNotEmpty(dar.getString(DarConstants.TRANSLATED_RESTRICTION)) ? dar.getString(DarConstants.TRANSLATED_RESTRICTION).replace("\n", " ") :  "";
        String rus = StringUtils.isNotEmpty( dar.getString(DarConstants.RUS)) ?  dar.getString(DarConstants.RUS).replace("\n", " ") : "";
        darWriter.write(
                dar.getString(DarConstants.DAR_CODE) + DEFAULT_SEPARATOR +
                        dataSetDetail.get(0).getString("name") + DEFAULT_SEPARATOR +
                        dar.get(DarConstants.DATASET_ID, ArrayList.class).get(0).toString() + DEFAULT_SEPARATOR +
                        consent.getName() + DEFAULT_SEPARATOR +
                        customContent1 +
                        sDUL + DEFAULT_SEPARATOR +
                        translatedRestriction + DEFAULT_SEPARATOR +
                        customContent2 +
                        formatTimeToDate(election.getFinalVoteDate().getTime()) + DEFAULT_SEPARATOR +
                        " " + END_OF_LINE);
    }

    public void addApprovedDARLine(FileWriter darWriter, Election election, Document dar, String profileName, String institution, Consent consent) throws IOException {
        String rus = StringUtils.isNotEmpty( dar.getString(DarConstants.RUS)) ?  dar.getString(DarConstants.RUS).replace("\n", " ") : "";

        String content1 = profileName + DEFAULT_SEPARATOR +
                institution;
        String content2 =                 rus + DEFAULT_SEPARATOR +
                formatTimeToDate(dar.getDate(DarConstants.SORT_DATE).getTime()) + DEFAULT_SEPARATOR +
                formatTimeToDate(election.getFinalVoteDate().getTime()) + DEFAULT_SEPARATOR;

        addDARLine(darWriter, election, dar, content1, content2, consent);
    }

    public void addReviewedDARLine(FileWriter darWriter, Election election, Document dar, Consent consent) throws IOException {
        String finalVote = election.getFinalVote() ? "Yes" : "No";
        addDARLine(darWriter, election, dar, finalVote, "", consent);
    }

```

Reply:
Thanks @ahaessly  I combined the common functionality, thanks for your suggestion!